server/src/main/java/io/crate/protocols/postgres/types/PGFloatVectorType.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.protocols.postgres.types;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.jetbrains.annotations.NotNull;
+
+import com.carrotsearch.hppc.ByteArrayList;
+
+import io.crate.protocols.postgres.parser.PgArrayParser;
+import io.netty.buffer.ByteBuf;
+
+public class PGFloatVectorType extends PGType<float[]> {
+
+    private static final byte[] NULL_BYTES = new byte[]{'N', 'U', 'L', 'L'};
+
+    public static final PGFloatVectorType INSTANCE = new PGFloatVectorType();
+    private static final int OID = 31;
+
+    PGFloatVectorType() {
+        super(OID, -1, -1, "floatvector");
+    }
+
+    @Override
+    public int typArray() {
+        return 1021;
+    }
+
+    @Override
+    public int typElem() {
+        return RealType.OID;
+    }
+
+    @Override
+    public String typeCategory() {
+        return TypeCategory.ARRAY.code();
+    }
+
+    @Override
+    public String type() {
+        return Type.BASE.code();
+    }
+
+    @Override
+    public int writeAsBinary(ByteBuf buffer, @NotNull float[] value) {
+        List<Object> list = new ArrayList<>(value.length);
+        for (float f : value) {
+            list.add(f);
+        }
+        return PGArray.FLOAT4_ARRAY.writeAsBinary(buffer, list);
```

Comment:
Would be good if we could avoid the boxing here, We know that the dimension is fixed and the inner type so the implementation should be a lot simpler than what `FLOAT4_ARRAY.writeAsBinary(buffer, list);` is doing.